### PR TITLE
Ldanzinger/sync

### PR DESCRIPTION
**Pull request by ldanzinger**
@khajra please review and merge. This is the edit and sync sample. I also fixed up some issues with the gradient in QML and the initializer list in clang (which was giving warnings due to ordering)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/EditAndSyncFeatures.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<html>
+	<img src="EditAndSyncFeatures.png"></img>
+	<hr>
+	<p>This sample demonstrates how to generate an offline mobile geodatabase from a feature service, edit the feature's geometry while disconnected, and synchronize the local edits back up to a feature service.</p>
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/EditAndSyncFeatures.html]: Remove the word "up".

File: ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#Edit and sync features
+
+This sample demonstrates how to generate an offline mobile geodatabase from a feature service, edit the feature's geometry while disconnected, and synchronize the local edits back up to a feature service.
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/README.md]: Remove the word "up".

File: ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+##How it works
+Pan and zoom into the desired area, making sure the area you want to take offline is within the red rectangle. Tap on the Generate Geodatabase button, and this will call `generateGeodatabase`, which will return a `GenerateGeodatabaseJob`. The job must be started manually, and once started can be canceled at anytime. Once the job completes successfully, a `FeatureLayer` is created with the output `FeatureTable` (which is obtained from the result `Geodatabase`), and the layer is added to the `Map`. Once the offline layer is displayed, you can select a feature by tapping on it, and move it to a new location by tapping anywhere on the map. Once there has been a successful edit, the sync button becomes enabled. This button will synchronize the local edits that have been made in the mobile geodatabase with the remote feature service. Click the sync button, and this will call `syncGeodatabase`, passing in the `SyncGeodatabaseParameters` and the `Geodatabase`. Similar to the generate process, this will return a job (`SyncGeodatabasJob`), which must be started manually. Once the job completes successfully, the local edits have been synchronized with the feature service.
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_CppSamples/EditData/EditAndSyncFeatures/README.md]: I suggest mixing it up a bit, using of the word "once" with the word "after" or "when" where you mean after or when.

File: ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/EditAndSyncFeatures.html [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<html>
+	<img src="EditAndSyncFeatures.png"></img>
+	<hr>
+	<p>This sample demonstrates how to generate an offline mobile geodatabase from a feature service, edit the feature's geometry while disconnected, and synchronize the local edits back up to a feature service.</p>
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/EditAndSyncFeatures.html]: Remove the word "up".

File: ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#Edit and sync features
+
+This sample demonstrates how to generate an offline mobile geodatabase from a feature service, edit the feature's geometry while disconnected, and synchronize the local edits back up to a feature service.
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/README.md]: Remove the word "up"

File: ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+##How it works
+Pan and zoom into the desired area, making sure the area you want to take offline is within the red rectangle. Tap on the Generate Geodatabase button, and this will call `generateGeodatabase`, which will return a `GenerateGeodatabaseJob`. The job must be started manually, and once started can be canceled at anytime. Once the job completes successfully, a `FeatureLayer` is created with the output `FeatureTable` (which is obtained from the result `Geodatabase`), and the layer is added to the `Map`. Once the offline layer is displayed, you can select a feature by tapping on it, and move it to a new location by tapping anywhere on the map. Once there has been a successful edit, the sync button becomes enabled. This button will synchronize the local edits that have been made in the mobile geodatabase with the remote feature service. Click the sync button, and this will call `syncGeodatabase`, passing in the `SyncGeodatabaseParameters` and the `Geodatabase`. Similar to the generate process, this will return a job (`SyncGeodatabasJob`), which must be started manually. Once the job completes successfully, the local edits have been synchronized with the feature service.
```
Comment (tdunn) [inline ArcGISRuntimeSDKQt_QMLSamples/EditData/EditAndSyncFeatures/README.md]: Same comment as before: vary the use of "once" and "after" or "when", as appropriate.

**Comment by khajra** [comment]
:+1:

**Pull request merged**